### PR TITLE
feat: support tools and tool_choice params (#31)

### PR DESCRIPTION
**Pull request by vladisavvv**

### CODE REVIEW

File: aidial_adapter_openai/utils/request_classifier.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from typing import Any
+
+
+def is_request_used_functions_or_tools(data: Any):
```
Comment (adubovik) [inline aidial_adapter_openai/utils/request_classifier.py]: 1/ Ungrammatical. Maybe `does_request_use_functions_or_tools`?
2/ I suggest to rename `data` argument to `request`. Data is too generic.

File: aidial_adapter_openai/utils/request_classifier.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        or "tools" in data
+        or "tool_choice" in data
+    ):
+        return True
+
+    if "messages" not in data:
+        return False
+
+    for message in data["messages"]:
+        role = message.get("role", None)
```
Comment (adubovik) [inline aidial_adapter_openai/utils/request_classifier.py]: Simply `message.get("role")`. It returns None if the key is missing.

Comment (vladisavvv) [reply aidial_adapter_openai/utils/request_classifier.py]: fixed

File: aidial_adapter_openai/utils/request_classifier.py (hunk 1 above)
Comment (vladisavvv) [reply aidial_adapter_openai/utils/request_classifier.py]: fixed two points

**Review by adubovik — APPROVED** [review]

**Pull request merged**